components/brave_wallet/browser/blockchain_images_source_base_unittest.cc

```diff
@@ -0,0 +1,111 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/blockchain_images_source_base.h"
+
+#include <memory>
+#include <string_view>
+#include <utility>
+
+#include "base/files/file_path.h"
+#include "base/memory/ref_counted_memory.h"
+#include "base/path_service.h"
+#include "base/test/bind.h"
+#include "base/test/task_environment.h"
+#include "brave/components/brave_wallet/browser/test_utils.h"
+#include "brave/components/brave_wallet/browser/wallet_data_files_installer.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_wallet {
+
+class TestBlockchainImagesSource : public BlockchainImagesSourceBase {
+ public:
+  explicit TestBlockchainImagesSource(const base::FilePath& base_path)
+      : BlockchainImagesSourceBase(base_path) {}
+  using BlockchainImagesSourceBase::GetMimeTypeForPath;
+  using BlockchainImagesSourceBase::StartDataRequestForPath;
+};
+
+class BlockchainImagesSourceBaseTest : public testing::Test {
+ public:
+  BlockchainImagesSourceBaseTest(const BlockchainImagesSourceBaseTest&) =
+      delete;
+  BlockchainImagesSourceBaseTest& operator=(
+      const BlockchainImagesSourceBaseTest&) = delete;
+
+ protected:
+  BlockchainImagesSourceBaseTest()
+      : task_environment_(base::test::TaskEnvironment::TimeSource::MOCK_TIME) {}
+  ~BlockchainImagesSourceBaseTest() override = default;
+
+  void SetUp() override {
+    // brave/test/data/brave_wallet/1.0.1
+    brave_wallet::SetLastInstalledWalletVersionForTest(base::Version("1.0.1"));
+    source_ = std::make_unique<TestBlockchainImagesSource>(
+        BraveWalletTestDataFolder());
```

Comment:
this should be using BraveWalletComponentsTestDataFolder (copy or move the relevant file)